### ISSUE TITLE
Bug: the result will be incorrect when call gin.Context.Writer.Status() in our custom gin middleware used after this(#46)

### ISSUE THREAD

**Issue by zhyee** [gin-contrib/timeout#47]
consider this case:

```go
func Timeout() gin.HandlerFunc {
	return timeout.New(timeout.WithTimeout(time.Second*1),
		timeout.WithHandler(func(ctx *gin.Context) {
			ctx.Next()
		}),
		timeout.WithResponse(func(ctx *gin.Context) {
			ctx.String(http.StatusRequestTimeout, "timeout")
		}),
	)
}

func Logger() gin.HandlerFunc {
	return func(c *gin.Context) {
		c.Next()
		// access the status we are sending
		status := c.Writer.Status() // Bug: in any middleware used after this middleware, the result will always be 200.
		log.Println(status)
	}
}

func main() {
	r := gin.New()
	r.Use(Timeout())
	r.Use(Logger())

	r.GET("/test", func(c *gin.Context) {
		c.Status(http.StatusInternalServerError)
	})
	// Listen and serve on 0.0.0.0:8080
	r.Run(":8080")
}
```

**Comment by zhyee** [comment]
#46

**Comment by fr-wentianqi** [comment]
same issue here, by moving your timeout middleware down underneath the logger middleware will solve the issue

**Comment by appleboy** [comment]
fix in #46
